.github/workflows/npmpublish-beta.yml

```diff
@@ -18,14 +18,18 @@ jobs:
           node-version: 12
       - run: npm ci
 
+    strategy:
+      matrix:
+        node-version: [10.x, 12.x, 13.x]
+
   publish-npm:
     needs: build
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
       - uses: actions/setup-node@v1
         with:
-          node-version: 12
+          node-version: ${{ matrix.node-version }}
```

Comment:
You defined the different node version for the publish job. This will result in the publish job running multiple time (3 time exactly for every version one). And the build job continues to run on nodev12 (only). We would want to run the build job on the node version matrix and if the build jobs succeeds for every node version the publish job should run once (with some predefined node version like 10 or so). Same applies for the npmpublish.yml

Reply:
Totally agree @Supereg.  Only `nodejs.yml` needs `matrix` since it's responsible for running the tests on all the different versions.  Ideally the publish wouldn't go through unless all of those tests pass

Reply:
> Ideally the publish wouldn't go through unless all of those tests pass

This automatically covered by the `needs: build`

Reply:
okay I have fixed this.

Reply:
okay I have fixed this.